opencog/cython/opencog/grounded_object_node.pyx

```diff
@@ -0,0 +1,48 @@
+from opencog.atomspace cimport cCreateGroundedObjectNode, cPythonGroundedObject
+from libcpp.memory cimport shared_ptr
+from libcpp.string cimport string
+from cython.operator cimport dereference as deref
+
+def createGroundedObjectNode(name, obj, atomspace):
+    cdef shared_ptr[cGroundedObject] o_ptr
+    cdef shared_ptr[cGroundedObjectNode] node_ptr
+    cdef string node_name = <bytes>(name.encode())
+    if obj is not None:
+        o_ptr.reset(new cPythonGroundedObject(<PyObject*>obj))
+        node_ptr.reset(new cGroundedObjectNode(node_name, o_ptr))
+    else:
+        node_ptr.reset(new cGroundedObjectNode(node_name))
+
+    return GroundedObjectNode(PtrHolder.create(<shared_ptr[void]&>node_ptr),
+                              atomspace)
+
+cdef class GroundedObjectNode(Atom):
+
+    def __init__(self, ptr_holder, atomspace):
+        super(GroundedObjectNode, self).__init__(ptr_holder, atomspace)
+
+    cdef cGroundedObjectNode* get_c_grounded_object_node_ptr(self):
+        return <cGroundedObjectNode*>(self.get_c_value_ptr().get())
+
+    def set_object(self, obj):
+        cdef shared_ptr[cGroundedObject] o_ptr
+        o_ptr.reset(new cPythonGroundedObject(<PyObject*>obj))
+        self.get_c_grounded_object_node_ptr().set_object(o_ptr)
```

Comment:
I think deference would work here instead of raw pointer:
 
deref(self.get_c_value_ptr()).set_object(o_ptr)